src/main/java/frc/robot/Subsystems/ShooterSystem.java

```diff
@@ -19,47 +19,63 @@ public class ShooterSystem extends Subsystem {
     private final double KI = 0;
     private final double KP = 0;
     private PidController pid;
-    private double ERROR = 0.05;
-    private static final double SPEED_POINT = 0.5; // TODO: Figure ideal value for this variable.
+    private double ERROR = 50;
+    private static final double SPEED_TARGET_SPEAKER = 4000;
+    private static final double SPEED_TARGET_AMP = 2000;
 
     public ShooterSystem(CANSparkMax master, CANSparkMax follower){
         this.master = master;
         this.follower = follower;
         this.encoder = (RelativeEncoder) master.getEncoder();
-        this.follower.follow(master, true); // TODO: Make sure this needs to be inverted.
+        this.follower.follow(master, false);
         this.pid = new PidController(RunningRobot.getControl().getClock(),
-                ()-> SmartDashboard.getNumber("KP", KP),
-                ()-> SmartDashboard.getNumber("KI", KI),
-                ()-> SmartDashboard.getNumber("KD", KD),
-                ()-> SmartDashboard.getNumber("KF", KF));
-        pid.setOutputLimit(-0.5,0.5);
+                ()-> {return SmartDashboard.getNumber("KP", KP);},
+                ()-> {return SmartDashboard.getNumber("KI", KI);},
+                ()-> {return SmartDashboard.getNumber("KD", KD); },
+                ()-> {return SmartDashboard.getNumber("KF", KF);});
+        pid.setOutputLimit(-0.002,1);
         pid.setTolerance(ERROR, Time.milliseconds(500));
         SmartDashboard.putNumber("KP", KP);
         SmartDashboard.putNumber("KI", KI);
         SmartDashboard.putNumber("KD", KD);
         SmartDashboard.putNumber("KF", KF);
+
+        master.setIdleMode(CANSparkBase.IdleMode.kCoast);
+        follower.setIdleMode(CANSparkBase.IdleMode.kCoast);
     }
 
-    public double getPID(){
+    public double getPID(double SPEED_POINT){
         double speed = pid.applyAsDouble(getSpeed(),SPEED_POINT);
 
         SmartDashboard.putNumber("Speed", speed);
+        SmartDashboard.putBoolean("isStopped",false);
         return speed;
     }
 
-    public void shoot(){
-        this.master.set(getPID());
+
+    public void shootSpeaker(){
+        this.master.set(-getPID(SPEED_TARGET_SPEAKER));
```

Comment:
Um a bit messy.. i'd rather you remove the getPID() function.
Just write in the shooter_speaker(){
Speed=pid....(getSpeed(), SPEED_TARGER..);
master.set(spees);

It's much clear this way.